packages/devtools_app/lib/src/screens/inspector_v2/layout_explorer/box/box.dart

```diff
@@ -441,3 +454,95 @@ class BoxChildVisualizer extends StatelessWidget {
     );
   }
 }
+
+/// Encapsulation of [widths] and [heights] for the layout.
+class LayoutWidthsAndHeights {
+  LayoutWidthsAndHeights({
+    required this.widths,
+    required this.heights,
+    required this.availableWidth,
+    required this.availableHeight,
+  })  : assert(widths.length == 3),
+        assert(heights.length == 3) {
+    _displayWidths = _simpleFractionalLayout(
+      availableSize: availableWidth,
+      sizes: widths,
+    );
+    _displayHeights = _simpleFractionalLayout(
+      availableSize: availableHeight,
+      sizes: heights,
+    );
+  }
+
+  final List<double> widths;
+  final List<double> heights;
```

Comment:
can you add dart doc explaining the content of these lists and how we expect the values to be indexed

Reply:
Rather than using lists, maybe we should be using records or simply passing these values directly into the constructor. I don't think there's much benefit to having these in lists, especially if we're just using them to group a fixed size set of values together for convenience.

Reply:
Switched to use a record (`WidgetSizes`) to keep track of the padding and widget width or height.